src/Config/Stage/Deploy/SearchEngine.php

```diff
@@ -3,18 +3,23 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
-namespace Magento\MagentoCloud\Process\Deploy\InstallUpdate\ConfigUpdate\SearchEngine;
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Stage\Deploy;
```

Comment:
I don't like that we mixing configuration for Ece-tools with a configuration for Magento.
Looks like this namespace `Magento\MagentoCloud\Config\Stage` is used only for configuration of ece-tools, let's move Magento-related configuration to a different namespace.

Reply:
ECE-Tools's configuration in `.magento.env.yaml` is a source of truth, namespace is pretty accurate

Reply:
I mean in `Magento\MagentoCloud\Config\Stage` namespace was responsible for configuration provided by users for each stage of the deployment process. Now it will contain configuration Classes which generates a configuration for Magento. 
I think such configuration should be in `Magento\MagentoCloud\Config\` but not in `Stage` namespace.

Reply:
Got it, fixed